administrator/components/com_finder/src/Indexer/Query.php

```diff
@@ -1346,8 +1357,22 @@ protected function getTokenData($token)
 		else
 		{
 			// Add the term to the query.
-			$query->where('(t.term = ' . $db->quote($token->term) . ' OR t.stem = ' . $db->quote($token->stem) . ')')
-				->where('t.phrase = 0')
+			if ($this->wordmode == 1)
+			{
+				$query->where('(t.term LIKE ' . $db->quote($token->term . '%') . ' OR t.stem LIKE ' . $db->quote($token->stem . '%') . ')')
+					->where('t.common = 0');
+			}
+			elseif ($this->wordmode == 2)
+			{
+				$query->where('(t.term LIKE ' . $db->quote('%' . $token->term . '%') . ' OR t.stem = ' . $db->quote('%' . $token->stem . '%') . ')')
+					->where('t.common = 0');
+			}
+			else
+			{
+				$query->where('(t.term = ' . $db->quote($token->term) . ' OR t.stem = ' . $db->quote($token->stem) . ')');
+			}
+
+			$query->where('t.phrase = 0')
```

Comment:
beside the constant use please already mentioned I would use less copy paste.

```suggestion

			$searchTerm = $token->term;
			$searchStem = $token->stem;
			
			if ($this->wordmode == 1)
			{
				$searchTerm .= '%';
				$searchStem .= '%';
			}
			elseif ($this->wordmode == 2)
			{
				$searchTerm = '% . $searchTerm . '%';
				$searchStem = '% . $searchStem . '%';
			}
			
			$query->where('(t.term = ' . $db->quote($searchTerm) . ' OR t.stem = ' . $db->quote($searchStem) . ')');
			$query->where('t.phrase = 0')
```

beside this, when you touch it it would make sense to convert it to use perpared statements